### DISCUSSION TITLE
How get donation data by id

### DISCUSSION THREAD

**Discussion by DoAgency** [impress-org/givewp#6846] [Developer Q&A] [answered]
Hello,
sorry but in documentation i can't find solution to my problem.
I need a method to get donation data by id.
For example in WooCommerce is:
$order = wc_get_order( $id_order );
$email = $order->email;

Let me know please
Thanks
Regards

**Comment by JoshuaHungDinh** [comment]
Hey @DoAgency, it may depend on what data you are looking for. 

For general information on a specific donation you can take advantage of the donation model. The donation model will give you access to the following properties

``` 
$properties = [
        'id' => 'int',
        'formId' => 'int',
        'formTitle' => 'string',
        'purchaseKey' => 'string',
        'donorIp' => 'string',
        'createdAt' => DateTime::class,
        'updatedAt' => DateTime::class,
        'status' => DonationStatus::class,
        'type' => DonationType::class,
        'mode' => DonationMode::class,
        'amount' => Money::class,
        'feeAmountRecovered' => Money::class,
        'exchangeRate' => ['string', '1'],
        'gatewayId' => 'string',
        'donorId' => 'int',
        'firstName' => 'string',
        'lastName' => 'string',
        'email' => 'string',
        'subscriptionId' => ['int', 0],
        'billingAddress' => BillingAddress::class,
        'anonymous' => ['bool', false],
        'levelId' => ['string', ''],
        'gatewayTransactionId' => 'string',
        'company' => 'string',
        'comment' => 'string',
    ];  
    
    Heres an example:
    
    $donation = Give\Donations\Models\Donation::find($id);
    $email = $donation->email;

**Comment by DoAgency** [comment]
Hi @JoshuaHungDinh,
thanks for your answer!

How can get custom value from donation?
For example I created in form donation this custom value:
![Senza titolo](https://github.com/impress-org/givewp/assets/38939209/d119fa15-ad2b-419c-a2c7-79d4a93aad94)
I can get that? This doesn't work:
$donation = Give\Donations\Models\Donation::find($id);
$customValue = $donation->codice_campagna;

Thanks
Reagrds

**Reply by JoshuaHungDinh** [reply, marked as answer]
@DoAgency To clarify the Donation model gives you access to the properties I listed above. If you are looking to retrieve payment data that does not belong to the Donation model you will need a different approach. 

It looks like you are using the Form Field Manger to create a custom text input. You can retrieve this custom value by searching the payment meta by the meta key.  ```give()->payment_meta->get_meta('donation_id', 'meta_key')```.

Heres an example using your provided picture :

```give()->payment_meta->get_meta($id, 'codice_campagna')```

**Comment by DoAgency** [comment]
Godmorning @JoshuaHungDinh , 
How can get custom value from donation? **Here I have Form hidden value!**
For example I created in form donation this custom value:
![Screenshot 2023-09-18 163920](https://github.com/impress-org/givewp/assets/38939209/99d76937-82e6-46aa-bcd3-14198927f66b)

Thanks
Regards

**Comment by DoAgency** [comment]
Someone can answer me please?
Thanks

**Reply by JoshuaHungDinh** [reply]
Hello @DoAgency, have you tried using the same method listed above? ```give()->payment_meta->get_meta('donation_id', 'meta_key')```

**Comment by DoAgency** [comment]
Hi @JoshuaHungDinh thanks for reply.
Yes I tried but that return undefined, maybe bacause I have "hidden form filed" and not "text field"
